app/controllers/AdminUsers/rewards_controller.rb

```diff
@@ -38,6 +40,37 @@ def destroy
       redirect_to admin_users_rewards_path, notice: 'Reward was successfully deleted.' if reward.destroy
     end
 
+    def new_import_from_csv
+      render :new_import_from_csv, locals: { reward: Reward.new }
+    end
+
+    def import_from_csv
+      if params[:reward][:file].nil?
+        redirect_to new_import_from_csv_admin_users_rewards_path, alert: 'Please select a file'
+      elsif File.extname(params[:reward][:file]) != '.csv'
+        redirect_to new_import_from_csv_admin_users_rewards_path, alert: 'Only .csv files are allowed.'
+      else
+        begin
+          ActiveRecord::Base.transaction do
```

Comment:
This could be moved to separate class (eg. service for importing).

Reply:
Done 👍 I have implemented seperate service for importing :)